packages/react-component-library/src/styled-components/partials/StyledOuterWrapper.tsx

```diff
@@ -0,0 +1,80 @@
+import styled, { css } from 'styled-components'
+import { selectors } from '@royalnavy/design-tokens'
+
+const { color } = selectors
+
+const DEFAULT_BORDER = css`
+  border: 1px solid ${color('neutral', '200')};
+`
+
+export interface StyledOuterWrapperProps {
+  $hasFocus?: boolean
+  $isDisabled?: boolean
+  $isInvalid?: boolean
+}
+
+function getBoxShadowColor({ $hasFocus }: StyledOuterWrapperProps) {
+  if ($hasFocus) {
+    return color('action', '100')
+  }
+
+  return 'transparent'
+}
+
+function getBorder({
+  $hasFocus,
+  $isDisabled,
+  $isInvalid,
+}: StyledOuterWrapperProps) {
+  if ($hasFocus) {
+    return css`
+      border: 3px solid ${color('action', '500')};
+    `
+  }
+
+  if ($isDisabled) {
+    return css`
+      border: 1px solid transparent;
+    `
+  }
+
+  if ($isInvalid) {
+    return css`
+      border: 3px solid ${color('danger', '800')};
+    `
+  }
+
+  return null
+}
+
+function getMargin({ $hasFocus, $isInvalid }: StyledOuterWrapperProps) {
+  if ($hasFocus || $isInvalid) {
+    return css`
+      margin: -2px;
+    `
+  }
+
+  return null
+}
+
+function getBackgroundColor({ $isDisabled }: StyledOuterWrapperProps) {
+  return css`
+    background-color: ${$isDisabled
+      ? color('neutral', '000')
+      : color('neutral', 'white')};
+  `
+}
+
+export const StyledOuterWrapper = styled.div<StyledOuterWrapperProps>`
+  ${(props) => {
+    return css`
+      ${getBackgroundColor(props)};
+      ${getBorder(props) || DEFAULT_BORDER};
+      ${getMargin(props)};
+      border-radius: 15px;
```

Comment:
Not one for now but maybe we should make some package wide tokens/constants for things like this border radius value?